src/components/Company/CompanyName/CorrectNameRequest.vue

```diff
@@ -0,0 +1,138 @@
+      filled
+      label="Enter the Applicant Email Address"
+      hint="Example: name@email.com"
+      persistent-hint
+      :rules="entityEmailRules"
+      data-test="entity-email"
+    >
+    </v-text-field>
+  </v-form>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Prop, Watch, Emit, Mixins } from 'vue-property-decorator'
+import { Action } from 'vuex-class'
+
+// Mixins
+import { NameRequestMixin } from '@/mixins'
+
+// Interfaces
+import { NrCorrectionIF } from '@/interfaces/correction-interfaces'
+import { ActionBindingIF } from '@/interfaces'
+
+@Component({})
+export default class CorrectNameRequest extends Mixins(NameRequestMixin) {
+  /** Trigger form submission */
+  @Prop({ default: false }) submit: boolean
+
+  @Action setNameRequest!: ActionBindingIF
+
+  private valid = false
+  private nameRequestNumber: string = ''
+  private entityPhone: string = ''
+  private entityEmail: string = ''
+
+  // Rules
+  private entityNumRules = [
+    v => !!v || 'Name Request Number is required',
+    v => this.validateNameRequestNumber(v) || 'Name Request Number is invalid'
+  ]
+  private entityPhoneNumberRules = [
+    v => this.isInputEntered(v, 'phone') || 'Phone number is required',
+    v => !(v.length > 12) || 'Phone number is invalid'
+  ]
+  private entityEmailRules = [
+    v => this.isInputEntered(v, 'email') || 'Email is required',
+    v => this.isValidateEmail(v) || 'Email is Invalid'
+  ]
+
+  // Validations
+  private get isFormValid (): boolean {
+    return this.valid && !!this.nameRequestNumber &&
+      (!!this.entityPhone || !!this.entityEmail)
+  }
+
+  private isInputEntered (value: any, inputType: string) {
+    return (!!((inputType === 'email') ? this.entityPhone : this.entityEmail) || !!value)
+  }
+
+  private isValidateEmail (value: any) {
+    return ((!!this.entityPhone && !!value) || !!this.validateEmailFormat(value))
+  }
+
+  private validateNameRequestNumber (value: string): boolean {
+    const VALID_FORMAT = new RegExp(/^(NR )?\d+$/)
+    return VALID_FORMAT.test(value.toUpperCase())
+  }
+
+  private validateEmailFormat (value: string): boolean {
+    const VALID_FORMAT = new RegExp(/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/)
+    return VALID_FORMAT.test(value)
+  }
+
+  /** Called when parent triggers form submission */
+  @Watch('submit')
+  private async onSubmit (): Promise<any> {
+    try {
+      const nameRequestResponse = await this.validateNameRequest(
+        this.nameRequestNumber, this.entityPhone, this.entityEmail
+      )
+
+      if (nameRequestResponse) {
+        const nrCorrection: NrCorrectionIF = {
+          nrNumber: this.nameRequestNumber,
+          legalName: this._getApprovedName(nameRequestResponse)
+        }
+        this.setNameRequest(nrCorrection)
+        this.emitSave()
+      }
+    } catch (e) {
+      this.emitDone()
```

Comment:
Error handling?
PS I think you can chain an error handler to an awaited method.

Reply:
oo right, k i will look at this!